### PR TITLE
PHOENIX-6325 Adapt queryserver build to the new phoenix-client artifactIds

### PR DESCRIPTION
**Pull request by stoty**

### CODE REVIEW

File: BUILDING.md [hunk 1]
```diff
@@ -29,40 +29,44 @@ By default, this tarball does not contain a Phoenix client jar as it is meant to
...
-for integration testing by activating the corresponding `hbase<minor.major>` profile.
+When specifying `phoenix.version` and `phoenix.client.artifactid`, also specify the HBase version to
+be used for integration testing by activating the corresponding `hbase<minor.major>` profile.
 
-When using a Phoenix 5.x version, activate the `hbase-2.x` profile in addition to the
+When using a Phoenix 5.1+ version, activate the `hbase-2.x` profile in addition to the
 profile for the minor 2.x version.
 
+**NOTE** that the integration tests cannot be currently run with Phoenix 4.16+ or 5.1+, so the
+examples below will fail. See https://issues.apache.org/jira/browse/PHOENIX-63241
```
Comment (richardantal) [inline BUILDING.md:63]: typo
I think it should be https://issues.apache.org/jira/browse/PHOENIX-6324

**Review by richardantal — COMMENTED** [review]
Tested the change and it seems to work.
I found a small typo in the BUILDING.md otherwise it looks good.

File: BUILDING.md (hunk 1 above)
Comment (stoty) [reply BUILDING.md:63]: Thank you @richardantal

Comment (stoty) [reply BUILDING.md]: Fixed

**Comment by stoty** [comment]
Hmm. for some reason Josh is listed as the author. I'm going to fix that.

**Review by joshelser — COMMENTED** [review]
Looks good!

**Pull request closed without merging**